module/src/main/scala/com/gu/googleauth/actions.scala

```diff
@@ -52,13 +50,54 @@ trait UserIdentifier {
     UserIdentity.fromRequest(request).filter(_.isValid || !authConfig.enforceValidity)
 }
 
-trait Actions extends UserIdentifier {
+object AuthAction {
+  type UserIdentityRequest[A] = AuthenticatedRequest[A, UserIdentity]
+}
+
+/**
+  * This action ensures that the user is authenticated and their token is valid. Is a user is not logged in or their
+  * token has expired then they will be authenticated.
+  *
+  * The AuthenticatedRequest will always have an identity.
+  *
+  * @param authConfig
+  * @param loginTarget The target that should be redirected to in order to carry out authentication
+  */
+class AuthAction[A](val authConfig: GoogleAuthConfig, loginTarget: Call)
+  (implicit bodyParser: BodyParser[A], val executionContext: ExecutionContext)
```

Comment:
If we're expecting people to create as part of their application wiring I can't see why it would need to be implicit. That said, my comment about "I feel like the library should provide this" may override this comment.